frontend/src/components/FormPanel.jsx

```diff
@@ -134,85 +49,201 @@ function Buttons({ onSave, onSubmit, approver }) {
+          onClick={onApprove}
+          className="btn btn-primary fw-bold me-2"
         >
           Mark approved
         </button>
       ) : (
         <button
           type="button"
           onClick={onSubmit}
-          className="btn btn-primary fw-bold"
+          className="btn btn-primary fw-bold me-2"
         >
           Submit to HQ
         </button>
       )}
+
+      {/* Show one loader for any loading state */}
+      <ButtonLoading show={loading} />
     </div>
   );
 }
 
 Buttons.propTypes = {
   onSave: PropTypes.func,
   onSubmit: PropTypes.func,
+  onApprove: PropTypes.func,
   approver: PropTypes.bool,
+  loading: PropTypes.bool,
 };
 
-function FormPanel({ show, setShow, formData, approver }) {
-  // Constants
-  const data = formData || {};
-  const currentYear = new Date().getFullYear();
-  const lastYear = currentYear - 1;
-
-  // States
-  const [park, setPark] = useState({
-    hasGate: false,
-    readyToPublish: false,
-  });
-  const [parkArea, setParkArea] = useState({
-    hasGate: false,
-    readyToPublish: false,
-  });
-  const [feature, setFeature] = useState({
-    hasGate: false,
-    readyToPublish: false,
-  });
-
+function SeasonForm({ seasonId, level, handleClose, onDataUpdate }) {
   // Hooks
-  const endpoint =
-    data.level && data.currentSeason?.id
-      ? `/seasons/${data.level}/${data.currentSeason.id}`
-      : null;
+  const { ROLES, checkAccess } = useAccess();
+  const approver = useMemo(
+    () => checkAccess(ROLES.APPROVER),
+    [checkAccess, ROLES.APPROVER],
+  );
+
+  const [data, setData] = useState(null);
+  const [notes, setNotes] = useState("");
+  const [deletedDateRangeIds, setDeletedDateRangeIds] = useState([]);
+
+  const { sendData: sendApprove, loading: sendingApprove } = useApiPost(
+    `/seasons/${seasonId}/approve/`,
+  );
+
+  const { sendData: sendSubmit, loading: sendingSubmit } = useApiPost(
+    `/seasons/${seasonId}/submit/`,
+  );
+
+  const { sendData: sendSave, loading: sendingSave } = useApiPost(
+    `/seasons/${seasonId}/save/`,
+  );
+
+  function addDeletedDateRangeId(id) {
+    setDeletedDateRangeIds((prev) => [...prev, id]);
+  }
 
   const {
-    data: seasonData = {},
-    // loading: seasonLoading,
-    // error: seasonError,
-  } = useApiGet(endpoint);
+    data: apiData,
+    loading,
+    error,
+    fetchData: refreshData,
+  } = useApiGet(`/seasons/${level}/${seasonId}`);
 
-  // Functions
-  function handleClose() {
-    setShow(false);
+  useEffect(() => {
+    if (apiData) {
+      setData(apiData);
+    }
+  }, [apiData]);
+
+  // Constants
+  const {
+    current: season,
+    previous: previousSeasonDates,
+    ...seasonMetadata
+  } = data || {};
+  const currentYear = season?.operatingYear;
+
+  // Clears and re-fetches the data
+  function resetData() {
+    setData(null);
+
+    // Refresh the data from the API
+    refreshData();
+  }
+
+  async function onSave(close = true) {
+    // Format the data for the API
```

Comment:
Some error handling would be nice in case, but it can be addressed in another ticket 👍